baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/AbstractDeprecatedApiCheck.java

```diff
@@ -73,11 +79,27 @@ private Description checkTree(Tree tree, VisitorState state) {
 
         Optional<Symbol> symbol = Optional.ofNullable(ASTHelpers.getSymbol(tree));
 
-        if (symbol.isPresent()) {
-            Optional<Name> currentClass = getCurrentClass(state);
-            if (currentClass.isPresent()
-                    && currentClass.get().equals(symbol.get().owner.getQualifiedName())) {
-                // Don't complain about deprecated APIs used within the same class
+        Optional<ClassSymbol> owningClass = getOwningClass(symbol);
+        Optional<URI> sourceFileUri = owningClass.map(c -> c.sourcefile).map(FileObject::toUri);
+        if (sourceFileUri.isPresent() && isRegularFileOnSystem(sourceFileUri.get())) {
+            // If the source file is a regular file on the local file system, this means we're calling a deprecated API
+            //   within the same project. We don't want to flag these usages, as they don't have any impact, and any
+            //   ABI break would have to be fixed immediately anyway.
+            // Note: This isn't triggered by files within the same repo for error-prone tests, because these use
+            //   in-memory file systems.
+            return Description.NO_MATCH;
+        }
+
+        // Note: the logic below will NOT work if the target dependency applies the "java" plugin rather than
+        //   the "java-library" plugin, because in that case the classfile that we get here will be within the
+        //   jar file itself, which makes it particularly tricky to distinguish from just regular jar dependencies.
+        // This should however be good enough for well-behaved repositories.
+        Optional<URI> classFileUri = owningClass.map(c -> c.classfile).map(FileObject::toUri);
+        if (classFileUri.isPresent() && isRegularFileOnSystem(classFileUri.get())) {
+            if (classFileUri.get().getPath().contains("/classes")) {
```

Comment:
Does this if need to be doubly nested? Just makes understanding the control flow a bit harder to understand I think.

Reply:
It does not - I've updated